internal/services/dns/dns_cname_record_resource.go

```diff
@@ -192,7 +192,7 @@ func resourceDnsCNameRecordRead(d *pluginsdk.ResourceData, meta interface{}) err
 			if props.TargetResource != nil && props.TargetResource.Id != nil {
 				targetResourceId = *props.TargetResource.Id
 			}
-			d.Set("target_resource_id", targetResourceId)
+			d.Set("target_resource_id", normalizeTargetResourceID(targetResourceId))
```

Comment:
We're only calling this once so it doesn't need to be a separate function. Can you please in-line the ID parsing above. We should also be raising the error from the parser instead of silently ignoring.